src/prefect/deployments/runner.py

```diff
@@ -351,8 +359,33 @@ async def apply(
                 trigger.set_deployment_id(deployment_id)
                 await client.create_automation(trigger.as_automation())
 
+            # We plan to support SLA configuration on the Prefect Server in the future.
+            # For now, we only support it on Prefect Cloud.
+            if self._sla:
+                await self._create_slas(deployment_id, client)
+
             return deployment_id
 
+    async def _create_slas(self, deployment_id: UUID, client: PrefectClient):
+        if not isinstance(self._sla, list):
+            self._sla = [self._sla]
+
+        if client.server_type == ServerType.CLOUD:
+            exceptions = []
+            for sla in self._sla:
+                try:
+                    sla.set_deployment_id(deployment_id)
+                    await client.create_sla(sla)
+                except Exception as e:
+                    e.add_note(f"SLA named '{sla.name}' failed to create")
```

Comment:
Unfortunately, `add_note` is only available in 3.11+ 😢. You'll need to raise another exception from the caught exception to add additional context.